vars/setupCommonPipelineEnvironment.groovy

```diff
@@ -35,19 +35,35 @@ void call(Map parameters = [:]) {
 
         def script = checkScript(this, parameters)
 
-        prepareDefaultValues script: script, customDefaults: parameters.customDefaults
+        String configFile = parameters.get('configFile')
+        loadConfigurationFromFile(script, configFile)
 
-        List customDefaults = ['default_pipeline_environment.yml'].plus(parameters.customDefaults?:[])
+        // Copy custom defaults from library resources to include them in the 'pipelineConfigAndTests' stash
+        List customDefaults = Utils.appendParameterToStringList(
+            ['default_pipeline_environment.yml'], parameters, 'customDefaults')
         customDefaults.each {
             cd ->
                 writeFile file: ".pipeline/${cd}", text: libraryResource(cd)
         }
 
-        stash name: 'pipelineConfigAndTests', includes: '.pipeline/**', allowEmpty: true
+        List customDefaultFiles = Utils.appendParameterToStringList(
+            [], parameters, 'customDefaultsFromFiles')
+        if (script.commonPipelineEnvironment.configuration.customDefaults) {
+            if (!script.commonPipelineEnvironment.configuration.customDefaults in List) {
+                // Align with Go side on supported parameter type.
+                error "You have defined the parameter 'customDefaults' in your project configuration " +
+                    "but it is of an unexpected type. Please make sure that it is a list of strings, i.e. " +
+                    "customDefaults = ['...']. See https://sap.github.io/jenkins-library/configuration/ for " +
```

Comment:
I don't find details there. Pending docs PR?

Reply:
Yes the docs still need to be updated. PR is not opened yet.

Reply:
I have not yet found this documented anywhere. What I can say for sure is that if you define "customDefaults" as string value, the pipeline will fail at the first Go step to the effect of "invalid config file format error". I wanted the Groovy side and Go side to behave the same here. Once I have adjusted the documentation, I would document this, also.